src/System.CommandLine/Binding/IBinder.cs

```diff
@@ -0,0 +1,15 @@
+﻿// Copyright (c) .NET Foundation and contributors. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System.CommandLine.Invocation;
+
+namespace System.CommandLine.Binding
+{
+    public interface IBinder
+    {
+        object GetTarget(InvocationContext context);
+        void SetTarget(object target);
+        void AddBinding(Binding binding);
+        void AddBinding(BindingSide targetSide, BindingSide parserSide);
```

Comment:
Overloads on an interface can generally be reduced to the base case, with the other(s) becoming extension methods.